glib/src/param_spec.rs

```diff
@@ -296,20 +296,13 @@ pub unsafe trait ParamSpecType:
 {
 }
 
-#[link(name = "gobject-2.0")]
-extern "C" {
-    pub static g_param_spec_types: *const ffi::GType;
-}
-
 macro_rules! define_param_spec {
-    ($rust_type:ident, $ffi_type:path, $rust_type_offset:expr) => {
-        // Can't use get_type here as this is not a boxed type but another fundamental type
+    ($rust_type:ident, $ffi_type:path, $type_name:literal) => {
         impl StaticType for $rust_type {
             #[inline]
             fn static_type() -> Type {
-                unsafe {
-                    from_glib(*g_param_spec_types.add($rust_type_offset))
-                }
+                const NAME: &CStr = unsafe { std::ffi::CStr::from_bytes_with_nul_unchecked(concat!($type_name, "\0").as_bytes()) };
+                unsafe { from_glib(gobject_ffi::g_type_from_name(NAME.as_ptr())) }
```

Comment:
```suggestion
                unsafe { from_glib(gobject_ffi::g_type_from_name(concat!($type_name, "\0").as_ptr() as *const _)) }
```

Also does the job btw. The divertion via `CStr` is not really necessary and just optimized away anyway here